tests/tensor/test_basic.py

```diff
@@ -3838,6 +3839,32 @@ def test_transpose():
     assert ptb.transpose(dmatrix()).name is None
 
 
+def test_matrix_transpose():
+    with pytest.raises(ValueError, match="Input array must be at least 2-dimensional"):
+        ptb.matrix_transpose(np.arange(6))
+
+    x2 = dmatrix("x2")
+    x3 = dtensor3("x3")
+
+    x2v = np.arange(6).reshape((2, 3))
+    x3v = np.arange(12).reshape((2, 3, 2))
+
+    f = pytensor.function(
+        [x2, x3],
+        [
+            ptb.matrix_transpose(x2),
+            ptb.matrix_transpose(x3),
+        ],
+    )
+    t2, t3 = f(x2v, x3v)
+
+    assert t2.shape == np.transpose(x2v).shape
+    assert np.all(t2 == np.transpose(x2v))
+    # TODO: Replace np.asarray([np.transpose(x3v[0]), np.transpose(x3v[1])]) with np.matrix_transpose(x3v) once numpy adds support for it (https://github.com/numpy/numpy/pull/24099)
+    assert t3.shape == np.asarray([np.transpose(x3v[0]), np.transpose(x3v[1])]).shape
+    assert np.all(t3 == np.asarray([np.transpose(x3v[0]), np.transpose(x3v[1])]))
```

Comment:
This is a very simple helper, instead of compiling and evaluating a whole function can you just `assert equal_computations([var], [expected_var])`. 

That will be explicit enough to test this functionality. 

Also check with the `.mt` helper

Reply:
Will it also make sense to replace the many assert statements in ```test_transpose``` with the helper ```assert equal_computations([var], [expected_var])```?
https://github.com/pymc-devs/pytensor/blob/bc3dda0ef9ea9d20934928ad0ef98ecd70eb705b/tests/tensor/test_basic.py#L3816-L3838

Reply:
We can leave the old tests the way they are, if they are not in the way